exp/services/ledgerexporter/exportmanager.go

```
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/stellar/go/ingest/ledgerbackend"
+	"github.com/stellar/go/xdr"
+)
+
+// LedgerCloseMetaObject represents a file with metadata and binary data.
+type LedgerCloseMetaObject struct {
+	// file name
+	objectKey     string
+	startSequence uint32
+	endSequence   uint32
+	// Actual binary data
+	data xdr.LedgerCloseMetaBatch
+}
+
+// AddLedgerCloseMeta adds a ledger
+func (f *LedgerCloseMetaObject) AddLedgerCloseMeta(ledgerCloseMeta xdr.LedgerCloseMeta) {
+	if f.startSequence == 0 {
+		f.data.StartSequence = xdr.Uint32(ledgerCloseMeta.LedgerSequence())
+	}
+	f.data.LedgerCloseMetas = append(f.data.LedgerCloseMetas, ledgerCloseMeta)
+	f.data.EndSequence = xdr.Uint32(ledgerCloseMeta.LedgerSequence())
+}
+
+// LedgerCount returns the number of ledgers added so far
+func (f *LedgerCloseMetaObject) LedgerCount() uint32 {
+	return uint32(len(f.data.LedgerCloseMetas))
+}
+
+type ExporterConfig struct {
+	LedgersPerFile    uint32 `toml:"ledgers_per_file"`
+	FilesPerPartition uint32 `toml:"files_per_partition"`
+}
+
+// getObjectKey generates the file name based on the ledger sequence.
+func (e *ExporterConfig) getObjectKey(ledgerSeq uint32) string {
+	var objectKey string
+
+	if e.FilesPerPartition != 0 {
+		partitionSize := e.LedgersPerFile * e.FilesPerPartition
+		partitionStart := (ledgerSeq / partitionSize) * partitionSize
+		partitionEnd := partitionStart + partitionSize - 1
+		objectKey = strconv.Itoa(int(partitionStart)) + "-" + strconv.Itoa(int(partitionEnd)) + "/"
+	}
+
+	// 0 ledgersPerFile is invalid, throw an error
```

nit: What you could do here is do a check at the start of the `getObjectKey` function and throw an explicit error or warning or something instead of doing a is_correct check and returning an empty or semi completed `objectKey` string.
Like in line 42 add
```
if e. LedgersPerFile == 0 {
  return objectKey, fmt.Error("LedgersPerFile is 0")
}
```

Not sure if you also want to change the returns to `string, error`

---

Valid point, but this is intended to be done during initial config validation. The comment was meant to be a to-do. I've now updated the comment indicating that.

---

Ah makes sense